lib/matplotlib/_api/deprecation.py

```diff
@@ -124,10 +124,9 @@ def deprecated(since, *, message='', name='', alternative='', pending=False,
     """
     Decorator to mark a function, a class, or a property as deprecated.
 
-    When deprecating a classmethod, a staticmethod, or a property, the
-    ``@deprecated`` decorator should go *under* ``@classmethod`` and
-    ``@staticmethod`` (i.e., `deprecated` should directly decorate the
-    underlying callable), but *over* ``@property``.
+    When deprecating a property, the ``@deprecated`` decorator should go *over*
+    ``@property``.  On the other hand ``@deprecated`` can be placed either over
+    or under ``@classmethod`` and ``@staticmethod``.
```

Comment:
What is the advantage of this?

Reply:
Just so that you don't have to remember where to put it?